routinator: update to 0.12.1 and add runit service.

klarasm opened the pull request:
Update to 0.12.0 and add runit service.

- [x] Create a runit service

No longer relevant for now: 
- Build and include documentation
- Add missing python packages required for documentation
    - python3-sphinx-tabs
    - python3-sphinx-copybutton

#### Testing the changes
- I tested the changes in this PR: **YES**
Release notes at https://github.com/NLnetLabs/routinator/releases

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc, x86_64-musl)
- I built this PR locally for these architectures:
  - aarch64-musl
  - armv7l
  - armv6l-musl
  - x86_64-musl

klarasm commented:
Fix a typo in INSTALL.msg, add second commit for documentation.

Duncaen commented:
Don't use `INSTALL.msg` for documentation add a `README.voidlinux` file like some other packages do if required.

klarasm commented:
Changed `INSTALL.msg` to `README.voidlinux`, fixed a build failure due to missing PyPI packages.

It seems that `vsv routinator` in `post_install()` fails. I'm a bit confused why this happens and haven't been able to replicate it on my build machine, even with `./xbps-src -K`.

Duncaen commented:
> It seems that `vsv routinator` in `post_install()` fails. I'm a bit confused why this happens and haven't been able to replicate it on my build machine, even with `./xbps-src -K`.

You did not commit the `srcpkgs/routinator/files/routinator` directory.

paper42 commented on srcpkgs/routinator/template:
```diff
@@ -12,15 +13,30 @@ changelog="https://raw.githubusercontent.com/NLnetLabs/routinator/main/Changelog
 esac
 
+post_build() {
+	pip install --user -r doc/manual/source/requirements.txt
```
use hostmakedepends, pip is not deterministic and will pollute the masterdir

klarasm replied:
What do I do if the packages are not in the void repos? I guess I could add them myself

klarasm commented:
> > It seems that `vsv routinator` in `post_install()` fails. I'm a bit confused why this happens and haven't been able to replicate it on my build machine, even with `./xbps-src -K`.
> 
> You did not commit the `srcpkgs/routinator/files/routinator` directory.

Oh, that was stupid of me. Will fix.

klarasm commented:
I've added the missing python packages, basically no testing done for them but the manual for routinator builds without pip now.

Is it ok for them to be in the same PR or should I create a separate one?

klarasm commented:
I will convert this to a regular PR as I can't think of anything else to do for now. I could drop the manual if the python dependencies are not desired and the manual is available at https://routinator.docs.nlnetlabs.nl anyway.

klarasm commented:
It seems that ARIN has changed their RPA so it doesn't require explicit acceptance anymore. This will be in the upcoming 0.12 release and will also remove the need for initialization to run the service:
[0.12.0-rc1 release](https://github.com/NLnetLabs/routinator/releases/tag/v0.12.0-rc1)
[Pull request detailing the change](https://github.com/NLnetLabs/routinator/pull/796).

This means that `README.voidlinux` can be removed after the release of 0.12.

klarasm commented:
Upstream has released 0.12.0. Rebase on that so I can drop `README.voidlinux`. Seems to be the same build failure as in #39516.

I dropped the manual as I'm now getting build failures with that and I'm not familiar enough with the python build process to correct that. I'll add the manual separately if I get to it.

klarasm commented:
Squashed commits and closed separate PR for update.

klarasm commented:
Upstream has released 0.12.1

paper42 commented on srcpkgs/routinator/files/routinator/run:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec routinator --config /etc/routinator/routinator.conf server --user=_routinator --group=_routinator 2>&1
```
If this logs something, having a logging service would be nice. Look at the dockerd service in the moby package for an example.


```suggestion
[ -r conf ] && . ./conf
exec routinator --config ${CONF_FILE:-/etc/routinator/routinator.conf} server --user=_routinator --group=_routinator $OPTS 2>&1
```

klarasm replied:
Good idea! I have included the suggestion in the new commit. Seems to work ok from what I can see.
